src/stm32f1/timer.cpp

```diff
@@ -377,35 +381,66 @@ hal::stm32f1::pwm general_purpose_timer_manager::acquire_pwm(timer_pins p_pin)
            p_pin };
 }
 
-hal::stm32f1::quadrature_encoder
-general_purpose_timer_manager::acquire_quadrature_encoder(timer_pins p_pin1,
-                                                          timer_pins p_pin2)
+template<typename... T>
+hal::v5::strong_ptr<hal::rotation_sensor> general_purpose_timer_manager::create(
+  std::pmr::polymorphic_allocator<> p_allocator,
+  T&&... args)
+{
+  return hal::v5::make_strong_ptr<hal::stm32f1::quadrature_encoder>(
+    p_allocator, quadrature_encoder::Key{ 0 }, ::std::forward<T>(args)...);
+}
+
```

Comment:
With the change I made before, you should be able to eliminate this and many of the other functions since the key is no longer needed.